compiler/rustc_builtin_macros/src/concat_bytes.rs

```diff
@@ -0,0 +1,210 @@
+                                    "cannot concatenate an array of string literals",
+                                );
+                                has_errors = true;
+                            }
+                            ast::LitKind::Float(_, _) => {
+                                cx.span_err(
+                                    expr.span,
+                                    "cannot concatenate an array of float literals",
+                                );
+                                has_errors = true;
+                            }
+                            ast::LitKind::Char(_) => {
+                                if let Ok(snippet) = cx.sess.source_map().span_to_snippet(expr.span)
+                                {
+                                    cx.struct_span_err(
+                                        expr.span,
+                                        "cannot concatenate an array of character literals",
+                                    )
+                                    .span_suggestion(
+                                        expr.span,
+                                        "try using a byte character",
+                                        format!("b{}", snippet),
+                                        Applicability::MachineApplicable,
+                                    )
+                                    .emit();
+                                } else {
+                                    cx.span_err(
+                                        expr.span,
+                                        "cannot concatenate an array of character literals",
+                                    )
+                                }
+                                has_errors = true;
+                            }
+                            ast::LitKind::Int(
+                                val,
+                                ast::LitIntType::Unsuffixed
+                                | ast::LitIntType::Unsigned(ast::UintTy::U8),
+                            ) if val <= 255 => {
+                                accumulator.push(val as u8);
+                            }
+                            ast::LitKind::Int(_, _) => {
+                                cx.span_err(expr.span, "numeric literal is out of bounds");
+                                has_errors = true;
+                            }
+                            ast::LitKind::Bool(val) => {
+                                cx.struct_span_err(
+                                    expr.span,
+                                    "cannot concatenate an array of boolean literals",
+                                )
+                                .span_suggestion(
+                                    expr.span,
+                                    "try using the numeric value",
+                                    (val as u8).to_string(),
+                                    Applicability::MachineApplicable,
```

Comment:
I think this is *probably* not a good suggestion to give; it seems unlikely that the user wanted `concat_bytes!(true)` to give them `[1]`. I would just leave the error.